### PR TITLE
Update TinyMCE to 4.6.x

### PR DESCRIPTION
**Pull request by iseulde**
This will need thorough testing.

Also `a[data-mce-selected]` and `code[data-mce-selected]` will have to be styled in the editor content.

Changelog: https://github.com/tinymce/tinymce-dist/blob/4.6.0/changelog.txt

Biggest features relevant for us:

* Inline caret boundaries for links and code elements.
* Internal paste flag.

### CODE REVIEW

**Comment by iseulde** [comment]
We can use the styles that @jasmussen created for Gutenberg: https://github.com/WordPress/gutenberg/blob/master/blocks/components/editable/style.scss

**Comment by azaozz** [comment]
TinyMCE 4.6.1 is out, many bugfixes: https://github.com/tinymce/tinymce-dist/commit/a4129cbcaea6d4e9c26b103f11e2215dd85b8cb7

**Comment by iseulde** [comment]
4.6.1 changelog: https://github.com/tinymce/tinymce-dist/blob/4.6.1/changelog.txt

**Comment by iseulde** [comment]
4.6.3 changelog: https://github.com/tinymce/tinymce-dist/blob/4.6.3/changelog.txt

This TinyMCE version is now well tested as part of WP core (currently at RC2).

**Comment by iseulde** [comment]
@jasmussen @folletto Could you have a look at the colours for the link boundaries? It looks like it's the same as WP core now, but it's not that nice on `:hover`, `:visited` etc. Also looks like it need same tweaking (negative margin, rounding).

**Comment by iseulde** [comment]
I'm wondering if we should keep the following styles. This is not _reading_ context.

```
a:link,
a:visited {
	color: $blue-medium;
}
a:hover,
a:active {
	color: $blue-light;
	outline: 0;
}
```

File: client/components/tinymce/iframe.scss [hunk 1]
```diff
@@ -185,3 +182,24 @@ hr {
...
 	background: lighten( $gray, 20 );
 }
+
+/* Styles for formatting the boundaries of anchors and code elements */
+.mce-content-body a[data-mce-selected] {
+	padding: 0 2px;
+	margin: 0 -2px;
+	border-radius: 2px;
+	box-shadow: 0 0 0 1px #bfe6ff;
+	background: #bfe6ff;
```
Comment (folletto) [inline client/components/tinymce/iframe.scss]: Shouldn't we be using our color variables here?

File: client/components/tinymce/iframe.scss [hunk 2]
```diff
@@ -185,3 +182,24 @@ hr {
...
+}
+
+.mce-content-body code {
+	padding: 2px 4px;
+	margin: 0;
+	border-radius: 2px;
+	color: #222;
+	background: #f2f4f5;
```
Comment (folletto) [inline client/components/tinymce/iframe.scss]: ...and here.

**Comment by folletto** [comment]
I think the color contrast is way way too low when the item is active:

screen shot 2017-06-06 at 11 01 59

It's hard if we want to keep the color of the link stable (maybe we can change the link color to be darker by default?). Here's an idea using `lighten( $gray, 30% )`:

cap-drop

**Comment by mtias** [comment]
Adjusted color when link is selected:

image

**Comment by mtias** [comment]
Adjusted to use color vars.

File: client/components/tinymce/iframe.scss [hunk 3]
```diff
@@ -26,13 +26,10 @@ q {
...
-}
+a:visited,
 a:hover,
 a:active {
-	color: $blue-light;
-	outline: 0;
+	color: $blue-medium;
 }
 
 code {
```
Comment (iseulde) [inline client/components/tinymce/iframe.scss]: It looks like some of these rules become redundant.

**Comment by mtias** [comment]
It is going in! Thanks @iseulde for setting this up.

**Pull request merged**